### PR TITLE
#668 remove custom stringify function in test

### PR DESCRIPTION
**Pull request by smellman**
#668 Remove custom `stringify` function in `test/lib/utils.ts`

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
Thanks for taking the time to open this PR!
There is a comment above the removed function about stringify of test cases results. 
Did you try and run the tests with UPDATE=1 and see which files change? 
Maybe we can save them after the regular stringify?

**Comment by smellman** [comment]
@HarelM Running UPDATE=1 npm run test generates many differences in JSON files.
I have committed these changes to the dev-668-update-1 branch.
https://github.com/smellman/maplibre-style-spec/commit/cd1574a7509e7eccc8e8b3a75113c9aed65cd8f8
Normally, UPDATE=1 is recognized as making destructive changes to make it harder to alter these files.

File: test/integration/expression/expression.test.ts [hunk 1]
```diff
@@ -52,7 +51,7 @@ describe('expression', () => {
                 delete fixture.metadata;
 
                 const fname = path.join(__dirname, expressionTestFileName);
-                fs.writeFileSync(fname, `${stringify(fixture)}\n`);
+                fs.writeFileSync(fname, JSON.stringify(fixture));
```
Comment (HarelM) [inline test/integration/expression/expression.test.ts:54]: This is writing everything is a single line, as can be seen in the linked commit.
I suggest to change it to:
```suggestion
                fs.writeFileSync(fname, JSON.stringify(fixture, null, 4));
```

**Comment by HarelM** [comment]
It's true that update=1 does destructive updates, but the difference between json compact and regular json stigngify should not be dramatic. 

It's also worth mentioning that we are using this library for the docs, so I'm not sure we will be able to remove it completely even if we remove this method. Just FYI.

File: test/integration/expression/expression.test.ts (hunk 1 above)
Comment (smellman) [reply test/integration/expression/expression.test.ts]: I use 2 space for format.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ## main
 
 ### ✨ Features and improvements
+- remove custom `stringify` function in test [#668](https://github.com/maplibre/maplibre-style-spec/issues/668))
```
Comment (HarelM) [inline CHANGELOG.md:4]: I didn't think it's interesting enough to be in the changelog...

**Comment by smellman** [comment]
I just upload https://github.com/smellman/maplibre-style-spec/commit/947ebcf07ee91ca009282933b6fe0e026b1538df with 2 space.
It may be a bit dramatic.

File: CHANGELOG.md (hunk 2 above)
Comment (smellman) [reply CHANGELOG.md:4]: OK, I will remove it.

**Comment by HarelM** [comment]
I'm trying to understand if we can use a `replacer` in the json stringify to support both these tests, but more importantly, the docs, which basically suffers from the same issue: 
Expression, which are basically an array, are expanded to new lines, which makes them harder to read.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#replacer

In any case, I went over the expression test files and they look ok to me. 

So feel free to research the above or commit the file changes as is.

**Comment by smellman** [comment]
My understanding is that the `replacer` is not same as the `formatter`.
Therefore, I have uploaded all converted files.

**Review by HarelM — APPROVED** [review]
Thanks!

**Pull request merged**